### ISSUE TITLE
Keep rules for Facebook SDK 5.5.1 are not sufficient

### ISSUE THREAD

**Issue by sgjesse** [facebook/facebook-android-sdk#583]
We have has a user reporting an issue on the Google R8 project that upgrading the Facebook SDK from 5.2.0. to 5.5.1 caused his app to break. This issue is here: https://issuetracker.google.com/141500096.

The failure happens in the reflection done here: https://github.com/facebook/facebook-android-sdk/blob/90d867fb0615dc95b8e74cf4aca20cd69a21203f/facebook-core/src/main/java/com/facebook/appevents/internal/InAppPurchaseEventManager.java#L203

I asked the user to generate the difference between the keep rules for these two versions (https://issuetracker.google.com/141500096#comment9), and the result is:

Version 5.5.1 has
```
# keep class names and method names used by reflection by InAppPurchaseEventManager
-keep,allowshrinking public class com.android.vending.billing.IInAppBillingService {
    public <methods>;
}

-keep,allowshrinking public class com.android.vending.billing.IInAppBillingService$Stub {
    public <methods>;
}
```
Version 5.2.0 has
```
-keep public class com.android.vending.billing.IInAppBillingService {
    public static com.android.vending.billing.IInAppBillingService asInterface(android.os.IBinder);
    public android.os.Bundle getSkuDetails(int,java.lang.String,java.lang.String,android.os.Bundle);
}
```
The problem with the rules used in version 5.5.1 is that `-keep,allowshrinking` (which is what `-keepnames` is synonym for) will not explicitly keep anything. It will only ensure that items (classes/methods/fields) which are otherwise retained through other use (e.g. explicit code use) will not be renamed. For items which are solely used through reflection such rule is not strong enough.

I suggest changing `-keepnames` to plain `-keep` in https://github.com/facebook/facebook-android-sdk/blob/a65fdc10c46487cbe6370ae0b7b7f30ea08208f3/facebook-core/proguard-rules.pro#L53. As I don't have any knowledge of the structure of the Facebook SDK code that rule might be too strong, but the current one is too weak - at least for some use cases.

**Comment by KylinChang** [comment]
The issue should be fixed in the next release

**Comment by KylinChang** [comment]
The issue should be addressed in v5.5.2
